### PR TITLE
backend:Implement Listing Issues Endpoints (LabelLab 13)

### PR DESCRIPTION
**Pull request by M-A-D-A-R-A**
# Description
Implement Listing Issues Endpoints

## Type of change

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

URL and Responses:- 
  - Fetch all Issue in a project :- `URL:- BASE_URL +/api/v1/issue/get/<int:project_id>`
     Example URL: `URL:- BASE_URL +/api/v1/issue/get/1`
     
 - Fetch Issue by Catergory:- `URL:- BASE_URL +/api/v1/issue/<int:project_id>/category/<string:category>`
    Example URL: `URL:- BASE_URL +/api/v1/issue/1/category/images`
     
 - Update Issue by team ID: -`URL:- BASE_URL +/api/v1/issue/<int:project_id>/team/<int:team_id>`
    Example URL: `URL:- BASE_URL +/api/v1/issue/1/team/1`
    
 - Fetch issue by Entity Type and ID:- `URL:- BASE_URL +/api/v1/issue/<int:project_id>/entity/<string:entity_type>/<int:entity_id>`
   Example URL: `URL:- BASE_URL +/api/v1/issue/1/entity/label/1` 
   

# Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass locally with my changes
- [ ] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

File: labellab-flask/api/controllers/issuecontroller.py [hunk 1]
```diff
@@ -106,15 +116,33 @@ class GetAllIssues(MethodView):
...
+    @project_member_only
     def get(self, project_id):
         try:
-                        
+            if not project_id:        
+                response = {
+                    "success": False,
+                    "msg": "Provide the project_id.",
+                }
+                return make_response(jsonify(response)), 200
```
Comment (amlannandy) [inline labellab-flask/api/controllers/issuecontroller.py:127]: Status code should not be 200. Can be 422 (Unprocessable Entry)

File: labellab-flask/api/controllers/issuecontroller.py [hunk 2]
```diff
@@ -106,15 +116,33 @@ class GetAllIssues(MethodView):
...
+
+            issues = find_all_issues_by_project_id(project_id)
+
+            if not issues:
+                response = {
+                        "success": False,
+                        "msg": "No Issue found",
+                        "body": {}
+                }
+                return make_response(jsonify(response)), 200
```
Comment (amlannandy) [inline labellab-flask/api/controllers/issuecontroller.py:137]: Here it should be 404 (Resource not found)

File: labellab-flask/api/controllers/issuecontroller.py [hunk 3]
```diff
@@ -314,16 +342,23 @@ class FetchCategoryIssuesView(MethodView):
...
+    @project_member_only
     def get(self,project_id, category):
         try:
-                        
+            if category not in allowed_categories:  
+                response = {
+                    "success": False,
+                    "msg": "Invalid category"
+                }
+                return make_response(jsonify(response)), 200
```
Comment (amlannandy) [inline labellab-flask/api/controllers/issuecontroller.py:353]: Again 422

File: labellab-flask/api/controllers/issuecontroller.py [hunk 4]
```diff
@@ -340,13 +375,49 @@ class FetchTeamIssuesView(MethodView):
      URL:- /api/v1/issue/<int:project_id>/team/<int:team_id>
     """
     @jwt_required
+    @project_member_only
     def get(self,project_id, team_id):
         try:
-                        
+            issue_team = fetch_all_issue_by_team_id(project_id,team_id)
```
Comment (amlannandy) [inline labellab-flask/api/controllers/issuecontroller.py:381]: Add the check to see if project_id and team_id are there. As you already did the check at one place

File: labellab-flask/api/controllers/issuecontroller.py [hunk 5]
```diff
@@ -314,16 +342,23 @@ class FetchCategoryIssuesView(MethodView):
      URL:- /api/v1/issue/<int:project_id>/category/<string:category>
     """
     @jwt_required
+    @project_member_only
     def get(self,project_id, category):
         try:
-                        
+            if category not in allowed_categories:
```
Comment (amlannandy) [inline labellab-flask/api/controllers/issuecontroller.py:348]: Validate project_id and category arguments

File: labellab-flask/api/controllers/issuecontroller.py [hunk 6]
```diff
@@ -340,13 +375,49 @@ class FetchTeamIssuesView(MethodView):
...
+    @jwt_required
+    @project_member_only
+    def get(self,project_id, entity_type, entity_id):
+        try:
+            if entity_type not in allowed_entity_types:
+                response = {
+                    'success': False,
+                    'msg': 'Invalid entity type'
+                }
+                return make_response(jsonify(response)), 400
```
Comment (amlannandy) [inline labellab-flask/api/controllers/issuecontroller.py:414]: Should be 422

File: labellab-flask/api/controllers/issuecontroller.py [hunk 7]
```diff
@@ -340,13 +375,49 @@ class FetchTeamIssuesView(MethodView):
...
+
+class FetchEntityIssuesView(MethodView):
+
+    """
+    This route fetches all project issue of a particular category
+     URL:- /api/v1/issue/<int:project_id>/entity/<string:entity_type>/<int:entity_id>
+    """
+    @jwt_required
+    @project_member_only
+    def get(self,project_id, entity_type, entity_id):
```
Comment (amlannandy) [inline labellab-flask/api/controllers/issuecontroller.py:407]: Input validation here also

**Review by amlannandy — CHANGES_REQUESTED** [review]

File: labellab-flask/api/controllers/issuecontroller.py (hunk 4 above)
Comment (M-A-D-A-R-A) [reply labellab-flask/api/controllers/issuecontroller.py:381]: How can i add a check for team_id?

Comment (amlannandy) [reply labellab-flask/api/controllers/issuecontroller.py:381]: Just check if team_id is provided or not. Similar to how you're checking project_id at line 122

**Review by amlannandy — APPROVED** [review]
Nice work. Thanks for the changes! 🍰

**Comment by amlannandy** [comment]
@M-A-D-A-R-A There are merge conflicts. Sync your fork and local with the organization repo. Then resolve the merge conflicts and push the changes.

**Comment by M-A-D-A-R-A** [comment]
> 

Yeah Sure

**Comment by amlannandy** [comment]
> > 
> 
> Yeah Sure

There are still some conflicts. Did you sync your branch with `upstream` ?

**Comment by M-A-D-A-R-A** [comment]
> > > 
> > 
> > 
> > Yeah Sure
> 
> There are still some conflicts. Did you sync your branch with `upstream` ?

Yeah I did Fetch the upstream, I tried merging master again in this branch there's no merge conflict.

Should I close this PR and make another one?

**Comment by amlannandy** [comment]
@M-A-D-A-R-A  No that should not be necessary. Can you go through all the files that you've made changes to in this PR and make sure there are no leftover conflicts in any of them? You've synced your fork with the org repo but you perhaps missed resolving the conflict.

**Comment by amlannandy** [comment]
Try out this from your current branch.

`git fetch upstream`

`git rebase upstream/master`

Then push the changes. Force push if necessary.

**Review by amlannandy — APPROVED** [review]

**Comment by M-A-D-A-R-A** [comment]
> Try out this from your current branch.
> 
> `git fetch upstream`
> 
> `git rebase upstream/master`
> 
> Then push the changes. Force push if necessary.

I tried these and made the necessary changes to each commit idk why it's saying merge conflicts.

**Comment by amlannandy** [comment]
> > Try out this from your current branch.
> > `git fetch upstream`
> > `git rebase upstream/master`
> > Then push the changes. Force push if necessary.
> 
> I tried these and made the necessary changes to each commit idk why it's saying merge conflicts.

Yeah this is weird. Let me check it out in the evening. Otherwise I'll merge it locally.

**Pull request closed without merging**